opennms-dao/src/main/java/org/opennms/netmgt/dao/support/AbstractTopLevelResourceType.java

```diff
@@ -47,7 +48,10 @@ public boolean isResourceTypeOnParent(OnmsResource parent) {
 
     @Override
     public List<OnmsResource> getResourcesForParent(OnmsResource parent) {
-        return getTopLevelResources();
+        if (parent == null) {
```

Comment:
That parameter is used by other resource types:
https://github.com/OpenNMS/opennms/blob/develop/opennms-dao/src/main/java/org/opennms/netmgt/dao/support/InterfaceSnmpResourceType.java#L117